lib/gooddata/lcm/lcm2.rb

```diff
@@ -17,36 +17,32 @@ class SmartHash < Hash
       def method_missing(name, *_args)
         key = name.to_s.downcase.to_sym
 
-        value = nil
-        keys.each do |k|
-          if k.to_s.downcase.to_sym == key
-            value = self[k]
-            break
-          end
-        end
-
-        if value
-          value
-        else
-          begin
-            super
-          rescue
-            nil
-          end
-        end
+        get_data(key)
       end
 
       def [](variable)
-        fail "Param #{variable} is not defined in the specification" if @specification && !@specification[variable] && !@specification[variable.to_sym]
-        super(variable)
+        get_data(variable)
       end
 
       def clear_filters
         @specification = nil
       end
 
       def setup_filters(filter)
-        @specification = filter
+        @specification = filter.to_hash
+      end
+
+      def check_specification(variable)
+        fail "Param #{variable} is not defined in the specification" if @specification && !@specification[variable.to_sym] && !@specification[variable.to_s]
+      end
+
+      def get_data(variable)
```

Comment:
The name get_data doesn't conform to the style-guide: https://github.com/bbatsov/ruby-style-guide#accessor_mutator_method_names